### DISCUSSION TITLE
Run in foreground with commandline flag

### DISCUSSION THREAD

**Discussion by benv666** [Chia-Network/chia-blockchain#5212] [Ideas]
Currently the chia node/farmer/etc all daemonize and fork to the background.
While this is great for certain applications, it's really annoying when trying to run chia as a supervised program (e.g. under supervisord or daemontools).
The current docker implementation has an ugly sleep loop, I presume that's because of this daemonization, but ideally I would like to see the docker run a single task (e.g. just the full node or only the farmer) in the foreground and either be pid 1, or run under supervisord so some other things like plotman can run next to it.

Suggestion: add -f flag to run in foreground that skips the daemonization. Bonus points for automatically logging to stdout/stderr.

**Comment by elmans1961** [comment]
I'm sorry but could you explain please, in what command you suggest flag.I'm new in programming and administration.

**Reply by benv666** [reply]
If you look for example at https://github.com/Chia-Network/chia-docker/blob/main/entrypoint.sh#L33 you see that the command is executed (and backgrounds) so the script continues and then needs to idle around (otherwise the docker dies) in line https://github.com/Chia-Network/chia-docker/blob/main/entrypoint.sh#L44

My proposal will allow the last line of this entrypoint to be something like:
`exec chia start farmer -f`

Or alternatively it can then be used in tools such as supervisord, see http://supervisord.org/subprocess.html#nondaemonizing-of-subprocesses for why the foreground flag is needed.

**Reply by elmans1961** [reply]
Thank you!
